plugins/bufferlines/bufferline.nix

```
@@ -125,196 +215,395 @@ in
+            matcher.__raw = # Lua
+              ''
+                function(buf)
+                  return buf.name:match('%.md') or buf.name:match('%.txt')
+                end
+              '';
+          }
+        ];
       };
-    in
-    mkIf cfg.enable {
-      extraPlugins = with pkgs.vimPlugins; [
-        cfg.package
-        nvim-web-devicons
+      indicator = {
+        style = "icon";
+        icon = "▎";
+      };
+      left_trunc_marker = "";
+      max_name_length = 18;
+      max_prefix_length = 15;
+      modified_icon = "●";
+      numbers.__raw = # Lua
+        ''
+          function(opts)
+            return string.format('%s·%s', opts.raise(opts.id), opts.lower(opts.ordinal))
+          end
+        '';
+      persist_buffer_sort = true;
+      right_trunc_marker = "";
+      separator_style = "slant";
+      show_buffer_close_icons = true;
+      show_buffer_icons = true;
+      show_close_icon = true;
+      show_tab_indicators = true;
+      sort_by = "extension";
+      tab_size = 18;
+      offsets = [
+        {
+          filetype = "neo-tree";
+          text = "File Explorer";
+          text_align = "center";
+          highlight = "Directory";
+        }
       ];
-      opts.termguicolors = true;
-      extraConfigLua = ''
-        require('bufferline').setup${helpers.toLuaObject setupOptions}
-      '';
     };
+    highlights =
+      let
+        commonBgColor = "#363a4f";
+        commonFgColor = "#1e2030";
+        commonSelectedAttrs = {
+          bg = commonBgColor;
+        };
+        selectedAttrsSet = builtins.listToAttrs (
+          map
+            (name: {
+              inherit name;
+              value = commonSelectedAttrs;
+            })
+            [
+              "buffer_selected"
+              "tab_selected"
+              "numbers_selected"
+              "close_button_selected"
+              "duplicate_selected"
+              "modified_selected"
+              "info_selected"
+              "warning_selected"
+              "error_selected"
+              "hint_selected"
+              "diagnostic_selected"
+              "info_diagnostic_selected"
+              "warning_diagnostic_selected"
+              "error_diagnostic_selected"
+              "hint_diagnostic_selected"
+            ]
+        );
+      in
+      selectedAttrsSet
+      // {
+        fill = {
+          bg = commonFgColor;
+        };
+        separator = {
+          fg = commonFgColor;
+        };
+        separator_visible = {
+          fg = commonFgColor;
+        };
+        separator_selected = {
+          bg = commonBgColor;
+          fg = commonFgColor;
+        };
+      };
+  };
+
+  extraConfig = cfg: {
+    extraConfigLua =
+      lib.mkIf cfg.diagnosticsUpdateInInsert # Lua
+        ''
+          vim.diagnostic.config { update_in_insert = true }
+        '';
+
+    extraPlugins = with pkgs.vimPlugins; [
+      cfg.package
+      nvim-web-devicons
+    ];
```

`cfg.package` is already added by `mkNeovimPlugin`
```suggestion
    extraPlugins = [ pkgs.vimPlugins.nvim-web-devicons ];
```

---

Thanks, missed that.